web/res/js/donationForm.js

```diff
@@ -331,8 +332,7 @@ $( function () {
 
 	function paymentDataIsValid() {
 		var currentState = store.getState();
-		return currentState.validity.amount ||
-			( currentState.donationFormContent.amount && currentState.donationFormContent.paymentType ) ;
+		return currentState.validity.amount;
```

Comment:
This might be misleading. It actually is the validity of payment data (including the payment type).